### ISSUE TITLE
[FEATURE] Proxy Support

### ISSUE THREAD

**Issue by nathom** [nathom/streamrip#518]
### Is the feature request related to a problem. Please describe it.

Sometimes TIDAL downloads fail after a while. May be fixed by rotating proxies.

### Describe the solution you would like.

Specify a list of proxies in the config file. Rotate proxies after a failure is detected.

**Comment by jieweix** [comment]
Same for Qobuz.

**Comment by NFEL** [comment]
Yeah, this would be an awesome feature that in my opinion can be added via config.toml file 

And most note that aiohttp does not support proxy by env [client_advanced](https://docs.aiohttp.org/en/stable/client_advanced.html#proxy-support).

and one last thing. since you are using requests to fetch some of api calls, it would be nicer to set proxy for all of them under a config attribute
